### PR TITLE
Issue#308 force autostart off when maint mode managed by CI

### PR DESCRIPTION
**Pull request by good-moon-oooo**
#308

### CODE REVIEW

File: classes/local/outage.php [hunk 1]
```diff
@@ -139,6 +140,12 @@ public function __construct($data = null) {
                 $this->$k = $v;
             }
         }
+
+        // If maintenance mode is managed by CI, set autostart to be off always.
+        if (file_exists("$CFG->dataroot/climaintenance.html")) {
+            $this->autostart = 0;
```
Comment (brendanheywood) [inline classes/local/outage.php]: there is a moodle form api method for locking a field, I *think* it is freeze but test this:

$mform->freeze('teamsubmission');

File: classes/local/outage.php [hunk 2]
```diff
@@ -139,6 +140,12 @@ public function __construct($data = null) {
                 $this->$k = $v;
             }
         }
+
+        // If maintenance mode is managed by CI, set autostart to be off always.
+        if (file_exists("$CFG->dataroot/climaintenance.html")) {
```
Comment (brendanheywood) [inline classes/local/outage.php]: also this isn't what I mean, I mean if the admin level config is set then do not allow it to be overridden at the form level

ie get_config for auth_outage/default_autostart

Comment (good-moon-oooo) [reply classes/local/outage.php]: Thanks for the further explanation, Brendan. I will change it not to be overridden at the form level if it has been set in admin settings.

File: classes/local/outage.php (hunk 1 above)
Comment (good-moon-oooo) [reply classes/local/outage.php]: Hi Brendan, Thanks for mentioning this function. This function works. I will change the code to it later.

**Pull request merged**